src/main/java/com/amplitude/Amplitude.java

```diff
@@ -81,12 +95,12 @@ public void useBatchMode(Boolean isBatchMode) {
   }
 
   private void updateHttpCall(HttpCallMode updatedHttpCallMode) {
-    if (httpCallMode == null || httpCallMode != updatedHttpCallMode) {
-      httpCallMode = updatedHttpCallMode;
+    httpCallMode = updatedHttpCallMode;
+    if (updatedHttpCallMode == HttpCallMode.BATCH_HTTPCALL) {
+      httpCall = new BatchHttpCall(apiKey, serverUrl != null ? serverUrl : Constants.BATCH_API_URL);
+    } else {
       httpCall =
-          (updatedHttpCallMode == HttpCallMode.BATCH_HTTPCALL)
-              ? new BatchHttpCall(apiKey)
-              : new GeneralHttpCall(apiKey);
+          new GeneralHttpCall(apiKey, serverUrl != null ? serverUrl : Constants.BATCH_API_URL);
```

Comment:
The default for general is not batch url right?

Reply:
Yes, the default is the general one.

Reply:
I mean when serverUrl is null here the fallback url is batch url not the general one.

Reply:
Oh, good catch! Will fix this.